File: leetcode/657.cpp
class Solution {
  public:
    bool judgeCircle(string moves) {
        int x = 0;
        int y = 0;

        for (auto c : moves) {
            if (c == 'U') { y += 1; }
            else if (c == 'D') { y -= 1; }
            else if (c == 'L') { x -= 1; }
            else if (c == 'R') { x += 1; }
        }

        return (x == 0) && (y == 0);
    }
};

File: leetcode/657.py
from collections import Counter

class Solution:
    def judgeCircle(self, moves: str) -> bool:
        counts = dict(Counter(moves))
        return (counts.get('L', 0) == counts.get('R', 0)) and \
               (counts.get('U', 0) == counts.get('D', 0))
        
